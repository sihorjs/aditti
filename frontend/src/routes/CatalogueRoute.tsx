import React, { Fragment } from "react";

import FeaturedProducts from "components/FeaturedProducts";
import Categories from "components/Categories";
import Products from "components/Products";

function CatalogueRoute() {
    return (
        <Fragment>
            <FeaturedProducts />
            <Categories />
            <Products />
        </Fragment>
    );
}

export default CatalogueRoute;
