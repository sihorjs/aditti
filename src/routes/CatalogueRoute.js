import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import Carousel from "components/Carousel";
import Categories from "components/Categories";
import ErrorMessage from "components/ErrorMessage";
import Catalogue from "components/Catalogue";
import Loading from "components/Loading";

import fetchCatalogue from "actions/catalogue";

class CatalogueRoute extends Component {
    componentDidMount() {
        this.props.onFetchCatalogue();
    }

    render() {
        if (this.props.isLoading) {
            return <Loading />;
        } else if (this.props.hasError) {
            return <ErrorMessage />;
        }

        return (
            <Fragment>
                <Carousel />
                <Categories />
                <Catalogue />
            </Fragment>
        );
    }
}

function mapStateToProps(state) {
    return {
        isLoading: state.catalogue.isLoading,
        hasError: state.catalogue.hasError
    };
}

function mapDispatchToProps(dispatch) {
    return {
        onFetchCatalogue: () => dispatch(fetchCatalogue())
    };
}

CatalogueRoute.propTypes = {
    isLoading: PropTypes.bool.isRequired,
    hasError: PropTypes.bool.isRequired,
    onFetchCatalogue: PropTypes.func.isRequired
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(CatalogueRoute);
