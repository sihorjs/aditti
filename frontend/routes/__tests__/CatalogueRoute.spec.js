import React from "react";
import { shallow } from "enzyme";
import CatalogueRoute from "routes/CatalogueRoute";

describe("<CatalogueRoute />", () => {
    it("renders without crash", () => {
        const tree = shallow(<CatalogueRoute />);

        expect(tree).toMatchSnapshot();
    });
});
