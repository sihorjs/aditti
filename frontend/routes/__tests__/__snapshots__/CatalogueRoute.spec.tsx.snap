// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogueRoute /> renders without crash 1`] = `ShallowWrapper {}`;
